app.js

```diff
@@ -1,162 +1,241 @@
 'use strict';
+
 var userInfo = {
   //temporary hard coded values
   "spiraUrl": "https://demo.spiraservice.net/rodrigo-pereira/",
   "username": "administrator",
   "apikey": "{AA50F584-BBC9-42A0-81BA-9F8A5CD8144A}",
+  "auth": "?username=administrator&api-key={AA50F584-BBC9-42A0-81BA-9F8A5CD8144A}",
+};
+
+var currentComponents = {
+  
+};
+
+var currentUsers = {
+  //relevant users will be stored here after selecting a project
 };
 
-var newRequirement = {
+var currentReleases = {
+  //relevant releases will be stored here after selecting a project
+};
+//template objects for correct order in sheet, currently requirement
+//template only
+
+var requirementType = {
+  "Package": -1,
+  "Need": 1,
+  "Feature": 2,
+  "Use Case": 3,
+  "User Story": 4,
+  "Quality": 5,
+  "Design Element": 6,
+};
+
+var requirementObj = {
   "RequirementId": null,
-  "Name": "test",
-  "Description": "test Description",
-  "ReleaseVersionNumber": null,
-  "RequirementTypeName": null,
-  "ImportanceName": null,
-  "StatusName": null,
+  "Name": null,
+  "Description": null,
+  "ReleaseId": null,
+  "RequirementTypeId": null,
+  "ImportanceId": null,
+  "StatusId": null,
   "EstimatePoints": null,
-  "AuthorName": null,
-  "OwnerName": null,
-  "ComponentName": null
+  "AuthorId": null,
+  "OwnerId": null,
+  "ComponentId": null,
 };
 
-(function () {
+var reqStatus = {
+  "Requested":	1,
+  "Evaluated":	7,
+  "Accepted":	5,
+  "Rejected":	6,
+  "Planned":	2,
+  "In Progress":	3,
+  "Developed":	4,
+  "Obsolete":	8,
+  "Tested":	9,
+  "Completed":	10,
+}
+
+//column ranges for different sheets, currently only requirements.
+var columnRanges = {
+  requirements: "A3:K"
```

Comment:
I need to keep the names the same as the artifacts because these are accessed through brackets (eg. columnRanges[artifact])